src/services/services.ts

```diff
@@ -7025,16 +7025,35 @@ namespace ts {
             }
 
             // TODO: add support for:
-            // - methods
-            // - constructors
-            // - class decls
-            let containingFunction = <FunctionDeclaration>getAncestor(tokenAtPos, SyntaxKind.FunctionDeclaration);
+            // - enums/enum members
+            // - interfaces
+            // - property declarations
+            // - potentially property assignments
+            let commentOwner: Node;
+            findOwner: for (commentOwner = tokenAtPos; commentOwner; commentOwner = commentOwner.parent) {
+                switch (commentOwner.kind) {
+                    case SyntaxKind.FunctionDeclaration:
+                    case SyntaxKind.MethodDeclaration:
+                    case SyntaxKind.Constructor:
+                    case SyntaxKind.ClassDeclaration:
+                    case SyntaxKind.VariableStatement:
+                        break findOwner;
+                    case SyntaxKind.SourceFile:
+                        return undefined;
+                    case SyntaxKind.ModuleDeclaration:
+                        // We don't want to give back a JSDoc for the 'b' in 'module a.b'.
+                        if (commentOwner.parent.kind === SyntaxKind.ModuleDeclaration) {

```

Comment:
No, if we hit one of the nested namespace declarations, then the position given is *within* the span of dotted names.

Check out `tests/cases/fourslash/docCommentTemplateNamespacesAndModules01.ts`